message PBEdge {
  required uint32 node0 = 1;
  required uint32 node1 = 2;

  repeated uint32 contained_faces = 3;
  repeated int32 contained_faces_chirality = 4;
  repeated uint32 contained_faces_eid = 5;
};

message PBFace {
  repeated uint32 nodes = 1;

  repeated uint32 edges = 2;
  repeated int32 edges_chirality = 3;

  required uint32 contained_cell0 = 4;
  required int32 contained_cell0_fid = 5;
  required uint32 contained_cell1 = 6;
  required int32 contained_cell1_fid = 7;
};

message PBCell {
  repeated uint32 nodes = 1;

  repeated uint32 faces = 2;
  repeated int32 faces_chirality = 3;

  repeated uint32 neighbor_cells = 4;
};

message PBMeshGraph {
  repeated PBEdge edges = 1;
  repeated PBFace faces = 2;
  repeated PBCell cells = 3;
};
